attestation-service/src/token/simple.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2023 by Alibaba.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::*;
+use base64::engine::general_purpose::URL_SAFE_NO_PAD;
+use base64::Engine;
+use rsa::pkcs1v15::SigningKey;
+use rsa::sha2::Sha384;
+use rsa::signature::{RandomizedSigner, SignatureEncoding};
+use rsa::traits::PublicKeyParts;
+use rsa::RsaPrivateKey;
+use serde_json::{json, Value};
+
+use crate::token::{AttestationTokenBroker, AttestationTokenConfig};
+
+const ISSUER_NAME: &str = "CoCo-Attestation-Service";
+const RSA_KEY_BITS: usize = 2048;
+const SIMPLE_TOKEN_ALG: &str = "RS384";
+
+pub struct SimpleAttestationTokenBroker {
+    private_key: RsaPrivateKey,
+    config: AttestationTokenConfig,
+}
+
+impl SimpleAttestationTokenBroker {
+    pub fn new(config: AttestationTokenConfig) -> Result<Self> {
+        let mut rng = rand::thread_rng();
+        let private_key = RsaPrivateKey::new(&mut rng, RSA_KEY_BITS)?;
+
+        Ok(Self {
+            private_key,
+            config,
+        })
+    }
+}
+
+impl SimpleAttestationTokenBroker {
+    fn rs384_sign(&self, payload: &[u8]) -> Result<Vec<u8>> {
+        let mut rng = rand::thread_rng();
+        let signing_key = SigningKey::<Sha384>::new(self.private_key.clone());
+        let signature = signing_key.sign_with_rng(&mut rng, payload);
+        Ok(signature.to_bytes().to_vec())
+    }
+}
+
+impl AttestationTokenBroker for SimpleAttestationTokenBroker {
+    fn issue(&self, custom_claims: Value) -> Result<String> {
+        let header_value = json!({
+            "typ": "JWT",
+            "alg": SIMPLE_TOKEN_ALG,
+        });
+        let header_string =
+            URL_SAFE_NO_PAD.encode(serde_json::to_string(&header_value)?.as_bytes());
+
+        let now = time::OffsetDateTime::now_utc();
+        let exp = now + time::Duration::minutes(self.config.duration_min);
+
+        let mut claims = json!({
+            "iss": ISSUER_NAME,
+            "jwk": serde_json::from_str::<Value>(&self.pubkey_jwks()?)?["keys"][0].clone(),
+            "nbf": now.unix_timestamp(),
+            "exp": exp.unix_timestamp(),
+        })
+        .as_object()
+        .unwrap()
+        .clone();
+
+        claims.extend(
+            custom_claims
+                .as_object()
+                .ok_or_else(|| anyhow!("Illegal token custom claims"))?
+                .to_owned(),
+        );
+
+        let claims_value = Value::Object(claims);
+        let claims_string =
+            URL_SAFE_NO_PAD.encode(&serde_json::to_string(&claims_value)?.as_bytes());
```

Comment:
Could you please bring `serde_json::to_string(&claims_value)?` outside to a single sentence? The nested callings is somehow hard to read